spec/confirm.ts

```diff
@@ -0,0 +1,127 @@
+    await confirm(user.email);
+    await User.findOne({ email: user.email }, (err, res) => {
+      assert.equal(true, res.confirmed);
+    });
+  });
+
+  it('sould be succes if user exist', done => {
+    chai
+      .request(app)
+      .post('/api/user/confirm')
+      .set('content-type', 'application/json')
+      .send({
+        token:
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLmNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4'
+      })
+      .end((err, res) => {
+        res.should.have.status(200);
+        assert.equal(
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb25maXJtZWQiOnRydWUsImVtYWlsIjoicHVwa2luQG1haWwuY29tIiwiaW1hZ2UiOiIiLCJpbnRlcmZhY2VMYW5nIjoiZW4iLCJuYW1lIjoiSXZhbiIsInN1YnNjcmlwdGlvbiI6dHJ1ZX0.gpgESzsa4rQJdHp4RvQ76U-ezVaFTVhyxjpSG8GqiOA',
+          res.body.token
+        );
+        done();
+      });
+  });
+
+  afterEach(async () => {
+    await User.remove({ email: 'pupkin@mail.com' });
+  });
+});
+
+describe('Token generation and decoding', () => {
+  it('shoul create correct token', () => {
+    const token = getToken({ email: 'pupkin@mail.com' });
+    assert.equal(
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLmNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4',
+      token
+    );
+  });
+
+  it('should decode token', () => {
+    const decoded = decodeToken(
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLmNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4'
+    );
+    assert.equal('pupkin@mail.com', decoded.email);
+  });
+});
+
+describe('Token generation and decoding', () => {
+  before(async () => {
+    const user = {
+      email: 'pupkin@mail.com',
+      interfaceLang: 'en',
+      name: 'Ivan',
+      password: 'Real Man'
+    };
+    const newcomer = new User(user);
+    await newcomer.save(err => {
+      return err;
+    });
+  });
+
+  it('shoult trow error if token invalid', done => {
+    chai
+      .request(app)
+      .post('/api/user/confirm')
+      .set('content-type', 'application/json')
+      .send({
+        token:
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLhNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4'
+      })
+      .end((err, res) => {
+        res.should.have.status(401);
+        done();
+      });
+  });
+
+  it('shoult return custom error if token invalid', done => {
+    chai
+      .request(app)
+      .post('/api/user/confirm')
+      .set('content-type', 'application/json')
+      .send({
+        token:
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLhNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4'
+      })
+      .end((err, res) => {
+        res.should.have.status(401);
+        assert.equal(104, res.body.error.code);
+        assert.equal('Token is invalid', res.body.error.enMessage);
+        assert.equal('Токен не валиден', res.body.error.ruMessage);
```

Comment:
This is not good. The server shouldn't care about the language.
Ideally, it should return just an error code or some kind of a constant (value from the enum)
Than the frontend decides how to handle the error and which message to display.